src/js/background/assignManager.js

```diff
@@ -197,12 +197,17 @@ window.assignManager = {
       return {};
     }
 
+    // proxyDNS only works for SOCKS proxies
+    if (result.proxy.type === "socks" || result.proxy.type === "socks4" ) {
```

Comment:
```suggestion
    if (["socks", "socks4"].includes(result.proxy.type)) {
```

Just a thought on the approach. Thanks for fixing this!

Reply:
Thanks for the suggestion, that is indeed a good idea!